File: FrogBaek/121_best_time_to_buy_and_sell_stock.cpp
class Solution
{
public:
    int maxProfit(vector<int>& prices) 
    {
        int max = 0, min = 0, res = 0;
        while (max < prices.size())
        {
            if (prices[max] - prices[min] > res)
                res = prices[max] - prices[min];

            if (prices[min] > prices[max])
                min = max;
            else
                max++;
        }

        return res;
    }
};

File: wch2208/121_Best_Time_to_Buy_and_Sell_Stock.js
/**
 * @param {number[]} prices
 * @return {number}
 */
var maxProfit = function (prices) {
  let answer = 0;
  let minPrice = Infinity;

  prices.forEach(currentPrice => {
    minPrice = Math.min(minPrice, currentPrice);
    const profit = currentPrice - minPrice;
    answer = Math.max(answer, profit);
  });

  return answer;
};

// Time taken: 31 m
